File: divide_Array_Into_Equal_Pairs/divide_Array_Into_Equal_Pairs.cpp


# include <iostream>
# include <vector>
# include <unordered_map>

using namespace std;

class Solution {
    public:
        static bool divideArray(vector<int>& nums) {
            std::unordered_map<int, int> freq;
    
            for ( int i = 0; i < nums.size(); i++ ) {
                freq[nums[i]]++;
            }
            for ( auto pair: freq ) {
                if ( pair.second & 1 != 0 ) return false; 
            }
            return true;
        }
    };

int main() {
    std::vector<int> nums = {1,1, 2};
    std::cout << Solution::divideArray( nums ) ;
}

File: divide_Array_Into_Equal_Pairs/divide_Array_Into_Equal_Pairs.py
from typing import List

class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        freq = {}
        i = 0
        while i < len( nums ) :
            if freq.get(nums[i], 0) == 0:
                freq[nums[i]] = 1
            else :
                freq[nums[i]] = freq[nums[i]] + 1
            i += 1
        for key in freq:
            if freq[key] & 1 != 0: return False
        return True

nums = [1,1]
solution = Solution()
print(solution.divideArray(nums))  